### PR TITLE
Feature: Global field localization using [lang-LANG]/index.json

### PR DESCRIPTION
**Pull request by grandsilence**
# Global field localization

In `[lang-LANG]/index.json`
1. Collection field name: `"field-[field_name]": "name",` (preferred)
2. Global field name for all collections: `"field-[collection]-[field_name]": "name",`

New helper method for it:
1. `$helpers.formatField(fieldAsObject)` where `field` is `object` with `field.field` and `field.collection`.
2. `$helpers.formatField(fieldName, collectionName)` where `fieldName` is `string` and `collectionName` is `string`.

### CODE REVIEW

**Comment by benhaynes** [comment]
Thank you @grandsilence!! We'll take a look asap!

**Comment by rijkvanzanten** [comment]
I'll try to get this in before the final release of V8.0.0 🚀

**Comment by everyx** [comment]
sorry, is this related #422 ?

**Comment by grandsilence** [comment]
Yeah, it’s implementation of #422

**Comment by everyx** [comment]
So we can set collection meta translate like this? https://github.com/directus/app/issues/422#issuecomment-455798477

image

**Comment by rijkvanzanten** [comment]
@everyx We might use the repeater interface here since we have that now, it's a little more user friendly than requiring raw json

File: src/helpers/format-field.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    (fieldIsString && !field.length)
+  ) {
+    console.error(`fieldLocalize: Invalid value of field: ${JSON.stringify(field)}`);
+    return "???";
+  }
+
+  const fieldName = fieldIsObject ? field.field : field;
+  const fallback = Vue.$helpers.formatTitle(fieldName);
+
+  if (!Vue.$te || !Vue.$t) {
```
Comment (rijkvanzanten) [inline src/helpers/format-field.js]: We're using this helper in the Directus app which will always have the Vue-i18n plugin available. This check is redundant

File: src/helpers/format-field.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  if (collectionName) {
+    const collectionField = `fields-${collectionName}-${fieldName}`;
+    if (Vue.$te(collectionField)) {
+      return Vue.$t(collectionField);
+    }
+  }
+
+  // Global field name (all collections)
+  const globalField = `fields-${fieldName}`;
```
Comment (rijkvanzanten) [inline src/helpers/format-field.js]: Is this how they're scoped in the app right now? 🤔 Maybe we should scope them as `collection.field` at all times. I can't think when we'd have a field name that should be global

File: src/helpers/format-field.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import Vue from "../main";
+
+export default function formatField(field, collection) {
```
Comment (rijkvanzanten) [inline src/helpers/format-field.js]: I'm wondering if we should make this more strict in only allowing field and collection to strings, and make them both required 🤔

**Pull request merged**